### PR TITLE
feat: module to withdraw from uniswap v2

### PR DESCRIPTION
**Pull request by gosuto-inzasheru**
tried deploying but brownie is being brownie:
```
Transaction sent: 0xecf13e6fa3b007a1398c56408105638a5eb791208ae13238e3e02af9b3c9d5a3
  Gas price: 0.0 gwei   Gas limit: 20000000   Nonce: 0
  UniswapWithdrawModule.constructor confirmed   Block: 8834175   Gas used: 699930 (3.50%)
  UniswapWithdrawModule deployed at: 0xfd7ba14DBDd791D6319147A87AE042C80F2B057a

  File "brownie/_cli/run.py", line 51, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 110, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/deploy_module_factory.py", line 7, in main
    return UniswapWithdrawModule.deploy(safe, {"from": deployer}, publish_source=True)
  File "brownie/network/contract.py", line 549, in __call__
    return tx["from"].deploy(
  File "brownie/network/account.py", line 557, in deploy
    contract.publish_source(deployed_contract, silent=silent)
  File "brownie/network/contract.py", line 359, in publish_source
    contract_info = self.get_verification_info()
  File "brownie/network/contract.py", line 316, in get_verification_info
    self._flattener = Flattener(source_fp, self._name, remaps, compiler_settings)
  File "brownie/project/flattener.py", line 30, in __init__
    self.traverse(primary_source_fp)
  File "brownie/project/flattener.py", line 71, in traverse
    self.traverse(import_path)
  File "brownie/project/flattener.py", line 50, in traverse
    source = fp_obj.read_text()
  File "/Users/gosuto/.pyenv/versions/3.9.16/lib/python3.9/pathlib.py", line 1266, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "/Users/gosuto/.pyenv/versions/3.9.16/lib/python3.9/pathlib.py", line 1252, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/Users/gosuto/.pyenv/versions/3.9.16/lib/python3.9/pathlib.py", line 1120, in _opener
    return self._accessor.open(self, flags, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/gosuto/Documents/safe_panic_modules/contracts/modules/interfaces/ISafe.sol'
Terminating local RPC client...
```

### CODE REVIEW

File: contracts/modules/UniswapWithdrawModule.sol [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    function uniswapV2Withdraw(address lp) external isSigner(safe) {
+        // https://docs.uniswap.org/contracts/v2/reference/smart-contracts/router-02#removeliquidity
+        IUniswapV2Pair ULP = IUniswapV2Pair(lp);
+        address token0 = ULP.token0();
+        address token1 = ULP.token1();
+
+        uint256 wethBefore = WETH.balanceOf(address(safe));
+
+        // assume withdrawal of total balance
+        uint256 ulpAmount = ULP.balanceOf(address(safe));
```
Comment (petrovska-petro) [inline contracts/modules/UniswapWithdrawModule.sol:65]: worth checking if this value is > 0 before proceeding with the rest of code

File: contracts/modules/UniswapWithdrawModule.sol [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        // if (wethAfter > wethBefore) {
+        //     _checkTransactionAndExecute(
+        //         safe,
+        //         address(WETH),
+        //         abi.encodeCall(IWETH9.withdraw, wethAfter - wethBefore)
+        //     );
+        // }
+    }
+
+    fallback() external payable {}
```
Comment (petrovska-petro) [inline contracts/modules/UniswapWithdrawModule.sol]: worth removing this i think if we cannot process weth -> eth smoothly

**Review by petrovska-petro — COMMENTED** [review]
overall looks good left minor comment missing hooking it up with the factory

also to help ups to verify the singleton worth adding the deployment script and the script that one of the signer could use for trigger the emerg. method in the uni module

File: contracts/modules/UniswapWithdrawModule.sol (hunk 1 above)
Comment (gosuto-inzasheru) [reply contracts/modules/UniswapWithdrawModule.sol:65]: 21de3ad

File: contracts/modules/UniswapWithdrawModule.sol (hunk 2 above)
Comment (gosuto-inzasheru) [reply contracts/modules/UniswapWithdrawModule.sol]: 21de3ad

**Review by petrovska-petro — APPROVED** [review]
alles goed! ✅

**Pull request merged**